File: longestConsecutiveSequence.js
class Solution {
  /**
   * @param {number[]} nums
   * @return {number}
   */
  longestConsecutive(nums) {
    let result = 0;

    let numSet = new Set(nums);

    for (const num of numSet) {
      if (!numSet.has(num - 1)) {
        let streak = 1;
        while (numSet.has(num + streak)) {
          streak++;
        }
        result = Math.max(result, streak);
      }
    }
    return result;
  }
}

File: longest_consecutive_sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0

        new_set = set(nums)

        for num in new_set:
            if num - 1 not in new_set:
                streak = 1
                while num + streak in new_set:
                    streak += 1
                result = max(result, streak)

        return result
